### DISCUSSION TITLE
Set Priority While Using Models

### DISCUSSION THREAD

**Discussion by Blair2004** [spatie/laravel-sitemap#412] [Q&A] [answered]
Hi, 
Based on this snippet : 

```php
use Spatie\Sitemap\Sitemap;

Sitemap::create()
    ->add($post)
    ->add(Post::all());
```
How can we setup the priority of the Post ? Is there any property that we need to add to the model ? Sorry if i haven't been able to spot that on the readme.

Thanks.

**Comment by HassanZahirnia** [comment, marked as answer]
Any news on this ?

Edit: Actually few minutes after posting this, I found the solution and it was pretty simple. 😄
In the model instead of returning a string, just return the Url object like so:
```php
public function toSitemapTag(): Url | string | array
{
   return Url::create( route('blog.post.show', $this) )
                ->setPriority(0.7);
}
```
